stomp_plugins/src/cost_functions/cartesian_distance.cpp

```diff
@@ -0,0 +1,220 @@
+    translation_tolerance_ = get_double_param(config, "translation_tolerance");
+    rotation_tolerance_ = get_double_param(config, "rotation_tolerance");
+
+    // Obtain tool link
+    tool_link_ = static_cast<std::string>(config["link_id"]);
+    if(!state_->knowsFrameTransform(tool_link_))
+    {
+      ROS_ERROR("Frame '%s' is not part of the model",tool_link_.c_str());
+      return false;
+    }
+
+    // total weight
+    cost_weight_ = position_cost_weight_ + orientation_cost_weight_;
+
+  }
+  catch(XmlRpc::XmlRpcException& e)
+  {
+    ROS_ERROR("%s failed to load parameters, %s",getName().c_str(),e.getMessage().c_str());
+    return false;
+  }
+
+  return true;
+}
+
+bool CartesianDistance::setMotionPlanRequest(const planning_scene::PlanningSceneConstPtr& planning_scene,
+                 const moveit_msgs::MotionPlanRequest &req,
+                 const stomp_core::StompConfiguration &config,
+                 moveit_msgs::MoveItErrorCodes& error_code)
+{
+  using namespace moveit::core;
+
+  bool found_goal = false;
+
+  // extracting goal joint values
+  for(const auto& gc : req.goal_constraints) {
+
+    // check joint constraints first
+    if (!gc.joint_constraints.empty()) {
+
+      // copying goal values into state
+      for (auto j = 0u; j < gc.joint_constraints.size(); j++) {
+        auto jc = gc.joint_constraints[j];
+        state_->setVariablePosition(jc.joint_name, jc.position);
+      }
+
+      // extract FK and break loop if goal is found
+      goal_ = state_->getFrameTransform(tool_link_);
+      found_goal = true;
+      break;
+    }
+
+    // check cartesian goal constraint
+    if (!gc.position_constraints.empty() && !gc.orientation_constraints.empty()) {
+        const auto pc = gc.position_constraints[0];
+        const auto oc = gc.orientation_constraints[0];
+
+        // assembling goal pose
+        Eigen::Quaterniond q = Eigen::Quaterniond::Identity();
+        geometry_msgs::Point p = pc.constraint_region.primitive_poses[0].position;
+        tf::quaternionMsgToEigen(oc.orientation, q);
+        goal_ = Eigen::Affine3d::Identity() * Eigen::Translation3d(Eigen::Vector3d(p.x, p.y, p.z)) * q;
+
+        // verify target frame
+        std::string frame_id = pc.header.frame_id;
+        if(!state_->knowsFrameTransform(frame_id))
+        {
+            ROS_ERROR("Frame '%s' is not part of the model",frame_id.c_str());
+            return false;
+        }
+
+        // transforming goal pose if target frame is differ
+        std::string target_frame = robot_model_->getModelFrame();
+        if(!frame_id.empty() && target_frame != frame_id)
+        {
+            Eigen::Affine3d root_to_frame = state_->getFrameTransform(frame_id);
+            Eigen::Affine3d root_to_target = state_->getFrameTransform(target_frame);
+            goal_ = (root_to_target.inverse()) * root_to_frame * goal_;
+        }
+
+        // break loop if goal is found
+        found_goal = true;
+        break;
+    }
+  }
```

Comment:
return is missing. Since you are returning bool it can cause undefined behavior

Reply:
This is not end of the method. return is on the line 152.